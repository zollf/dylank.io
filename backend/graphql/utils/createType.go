package utils

import "github.com/graphql-go/graphql"

var _cachedListType map[string]*graphql.List
var _cachedObjectType map[string]*graphql.Object

// Creates a cached type, useful for wrapper graphql functions that return type
func CreateList(key string, interfaceConfig *graphql.List) *graphql.List {
	if _cachedListType == nil {
		_cachedListType = make(map[string]*graphql.List)
	}

	if config, ok := _cachedListType[key]; ok {
		return config
	} else {
		_cachedListType[key] = interfaceConfig
		return _cachedListType[key]
	}
}

func CreateObject(key string, interfaceConfig *graphql.Object) *graphql.Object {
	if _cachedObjectType == nil {
		_cachedObjectType = make(map[string]*graphql.Object)
	}

	if config, ok := _cachedObjectType[key]; ok {
		return config
	} else {
		_cachedObjectType[key] = interfaceConfig
		return _cachedObjectType[key]
	}
}
