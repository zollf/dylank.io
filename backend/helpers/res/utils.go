package res

import (
	"fmt"

	"github.com/kataras/iris/v12"
)

func Redirect(ctx iris.Context, params string) bool {
	if redirect := ctx.FormValue("redirect"); redirect != "" {
		ctx.Redirect(fmt.Sprintf("%s%s", redirect, params))
		return true
	}
	return false
}

func AcceptsJSON(ctx iris.Context) bool {
	return ctx.GetHeader("Accepts") == "application/json"
}
