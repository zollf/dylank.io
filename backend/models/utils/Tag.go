package utils

import (
	"app/models/projects"
	"app/models/tags"
	"sort"
	"time"
)

type TagInterface struct {
	ID        uint64    `json:"id"`
	Slug      string    `json:"slug"`
	Title     string    `json:"title"`
	CreatedAt time.Time `json:"createdAt"`
	UpdatedAt time.Time `json:"updatedAt"`
	Count     int       `json:"count"`
}

type TagData struct {
	ID        uint64
	Index     int
	Title     string
	Slug      string
	CreatedAt string
	UpdatedAt string
}

type ProjectTagData struct {
	Title   string
	Slug    string
	Checked bool
}

// Given a tag, check it exists amongst an array of tags
func CheckTagExistsInTags(needle *tags.Tag, haystack []*tags.Tag) bool {
	for _, needleInHay := range haystack {
		if needleInHay.ID == needle.ID {
			return true
		}
	}
	return false
}

func TagOccurrencesInProjects(projects []*projects.Project, tag *tags.Tag) int {
	count := 0
	for _, project := range projects {
		for _, projectTag := range project.Tags {
			if projectTag.ID == tag.ID {
				count = count + 1
			}
		}
	}

	return count
}

func TagsOccurrencesInProjects(projects []*projects.Project) []*TagInterface {
	var tags []*TagInterface
	tagsMap := make(map[string]*TagInterface)
	for _, project := range projects {
		for _, tag := range project.Tags {
			if tagInterface, ok := tagsMap[tag.Slug]; ok {
				tagInterface.Count = tagInterface.Count + 1
			} else {
				tagsMap[tag.Slug] = &TagInterface{
					ID:        tag.ID,
					Slug:      tag.Slug,
					Title:     tag.Title,
					CreatedAt: tag.CreatedAt,
					UpdatedAt: tag.UpdatedAt,
					Count:     1,
				}
			}
		}
	}

	keys := make([]string, 0, len(tagsMap))
	for key := range tagsMap {
		keys = append(keys, key)
	}
	sort.Strings(keys)

	for _, key := range keys {
		tags = append(tags, tagsMap[key])
	}

	return tags
}

func GetTagsData() ([]*TagData, error) {
	tags, t_err := tags.All()
	if t_err != nil {
		return nil, t_err
	}

	var tag_data []*TagData
	zone, _ := time.LoadLocation("Australia/Perth")

	for i, tag := range tags {
		tag_data = append(tag_data, &TagData{
			ID:        tag.ID,
			Index:     i + 1,
			Title:     tag.Title,
			Slug:      tag.Slug,
			CreatedAt: tag.CreatedAt.In(zone).Format(time.RFC822),
			UpdatedAt: tag.UpdatedAt.In(zone).Format(time.RFC822),
		})
	}

	return tag_data, nil
}

func GetTagData(id string) (*TagData, error) {
	tag, not_found := tags.Find(id)

	if not_found != nil {
		return nil, not_found
	}

	tag_data := &TagData{
		ID:    tag.ID,
		Title: tag.Title,
		Slug:  tag.Slug,
	}

	return tag_data, nil
}
