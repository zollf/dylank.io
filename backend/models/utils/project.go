package utils

import (
	"app/models/assets"
	"app/models/projects"
	"app/models/tags"
	"time"
)

// Used for dom rendering
type ProjectData struct {
	ID          uint64
	Index       int
	Title       string
	Description string
	CreatedAt   string
	UpdatedAt   string
}

type ProjectDataWithTagAndAsset struct {
	ID          uint64
	Title       string
	Slug        string
	Description string
	Assets      []*assets.Asset
	URL         string
	Git         string
	Tags        []*tags.Tag
	CreatedAt   string
	UpdatedAt   string
}

func GetProjectData() ([]*ProjectData, error) {
	projects, projects_err := projects.All()
	if projects_err != nil {
		return nil, projects_err
	}

	var project_data []*ProjectData
	zone, _ := time.LoadLocation("Australia/Perth")

	for i, project := range projects {
		project_data = append(project_data, &ProjectData{
			ID:          project.ID,
			Index:       i + 1,
			Title:       project.Title,
			Description: project.Description,
			CreatedAt:   project.CreatedAt.In(zone).Format(time.RFC822),
			UpdatedAt:   project.UpdatedAt.In(zone).Format(time.RFC822),
		})
	}

	return project_data, nil
}

func GetProjectDataWithTagAndAsset(id string) (*ProjectDataWithTagAndAsset, []*ProjectTagData, []*ProjectAssetData, error) {
	project, project_err := projects.Find(id)

	if project_err != nil {
		return nil, nil, nil, project_err
	}

	zone, _ := time.LoadLocation("Australia/Perth")

	Git := ""
	if project.Git != nil {
		Git = *project.Git
	}

	URL := ""
	if project.URL != nil {
		URL = *project.URL
	}

	project_data := &ProjectDataWithTagAndAsset{
		ID:          project.ID,
		Title:       project.Title,
		Slug:        project.Slug,
		Assets:      project.Assets,
		URL:         URL,
		Git:         Git,
		Tags:        project.Tags,
		Description: project.Description,
		CreatedAt:   project.CreatedAt.In(zone).Format(time.RFC822),
	}

	var tags_data []*ProjectTagData
	tags, _ := tags.All()
	for _, tag := range tags {
		tags_data = append(tags_data, &ProjectTagData{
			Title:   tag.Title,
			Slug:    tag.Slug,
			Checked: CheckTagExistsInTags(tag, project.Tags),
		})
	}

	var assets_data []*ProjectAssetData
	assets, _ := assets.All()
	for _, asset := range assets {
		assets_data = append(assets_data, &ProjectAssetData{
			Title:   asset.Title,
			Slug:    asset.Slug,
			Url:     asset.Url,
			Checked: CheckAssetExistsInAssets(asset, project.Assets),
		})
	}

	return project_data, tags_data, assets_data, nil
}
