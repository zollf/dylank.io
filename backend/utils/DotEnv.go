package utils

import (
	"fmt"
	"log"

	"github.com/joho/godotenv"
)

func DotEnv() {
	root := RootDir()
	err := godotenv.Load(fmt.Sprintf("%s/.env", root))
	log.Printf("Initialising dotenv")

	if err != nil {
		log.Printf("Failed to initialising dotenv")
	}
}
