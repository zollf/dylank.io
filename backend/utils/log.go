package utils

import (
	"os"

	"github.com/sirupsen/logrus"
)

var _log *logrus.Logger

func Log() *logrus.Logger {
	if _log == nil {
		_log = logrus.New()
		_log.SetFormatter(&logrus.TextFormatter{
			ForceColors:   true,
			FullTimestamp: true,
		})
		_log.SetOutput(os.Stdout)
	}
	return _log
}
