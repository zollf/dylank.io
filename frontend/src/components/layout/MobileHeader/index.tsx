import React, { useCallback, useState } from 'react';
import Burger from '@/images/burger.svg';
import Close from '@/images/close.svg';

import MobileMenu from '../MobileMenu';
import styles from './styles.module.scss';

const MobileHeader = () => {
  const [active, setActive] = useState(false);
  const [delayClosed, setDelayedClosed] = useState(true);

  const toggle = useCallback(() => {
    active ? close() : open();
  }, [active]);

  const open = useCallback(() => {
    setDelayedClosed(false);
    setTimeout(() => setActive(true), 50);
  }, [setDelayedClosed, setActive]);

  const close = useCallback(() => {
    setActive(false);
    setTimeout(() => setDelayedClosed(true), 200);
  }, [setActive, setDelayedClosed]);

  return (
    <>
      <header className={styles.mobileHeader}>
        <div className={styles.logo}>d.io</div>
        <button type="button" className={styles.menu} onClick={toggle} data-active={active} data-testid="hbtn">
          {active ? <Close /> : <Burger />}
        </button>
        <MobileMenu active={active} />
      </header>
      {!delayClosed && (
        <button type="button" className={styles.backing} onClick={close} data-active={active} data-testid="backing" />
      )}
    </>
  );
};

export default MobileHeader;
