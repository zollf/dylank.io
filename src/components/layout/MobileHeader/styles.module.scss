.mobileHeader {
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  display: flex;
  height: $mobileHeaderHeight;
  background: $color-gray-colors-white;
  justify-content: space-between;
  padding: $margin * 0.5;
  z-index: 1;

  .logo {
    color: $color-brand-colors-secondary-1;

    @include text-style-ubuntu-medium-20-font-size;
  }

  .menu {
    @include ClearButton;

    cursor: pointer;
    transform: rotate(0);
    transition: 0.2s all ease;

    &[data-active="true"] {
      transform: rotate(180deg);
    }
  }
}

.backing {
  @include ClearButton;

  z-index: 0;
  position: fixed;
  top: 0;
  left: 0;
  height: 100%;
  width: 100%;
  background: rgba(black, 0.3);
  transition: 0.2s all ease;
  opacity: 1;
  cursor: pointer;

  &[data-active="false"] {
    opacity: 0;
  }
}