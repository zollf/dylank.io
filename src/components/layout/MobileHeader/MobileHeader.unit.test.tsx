import React from 'react';
import UserEvent from '@testing-library/user-event';
import { act, render, waitFor, waitForElementToBeRemoved } from '@testing-library/react';
import { setMobile } from '@/tests/utils';

import MobileHeader from './';

jest.useFakeTimers();

it('matches its snapshot', () => {
  const { baseElement } = render(<MobileHeader />);
  expect(baseElement).toMatchSnapshot();
});

it('matches its mobile snapshot', () => {
  setMobile();
  const { baseElement } = render(<MobileHeader />);
  expect(baseElement).toMatchSnapshot();
});

it('opens and close mobile menu correctly', async () => {
  setMobile();
  const { getByTestId } = render(<MobileHeader />);
  UserEvent.click(getByTestId('hbtn'));
  await waitFor(() => expect(getByTestId('backing')).toBeInTheDocument());

  act(() => {
    jest.runAllTimers();
  });

  UserEvent.click(getByTestId('hbtn'));
  await waitForElementToBeRemoved(() => getByTestId('backing'));
});
