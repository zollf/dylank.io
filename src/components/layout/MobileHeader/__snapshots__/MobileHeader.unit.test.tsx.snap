// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches its mobile snapshot 1`] = `
<body>
  <div>
    <header
      class="mobileHeader"
    >
      <div
        class="logo"
      >
        d.io
      </div>
      <button
        class="menu"
        data-active="false"
        data-testid="hbtn"
        type="button"
      >
        <svg
          fill="none"
          height="24"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <circle
            cx="12"
            cy="12"
            fill="#116ACC"
            r="12"
          />
          <path
            clip-rule="evenodd"
            d="M6 9.333V8h12v1.333H6zm0 3.334h12v-1.334H6v1.334zM8 16h10v-1.333H8V16z"
            fill="#fff"
            fill-rule="evenodd"
          />
        </svg>
      </button>
      <div
        class="mobileMenu"
        data-active="false"
        data-testid="mobileMenu"
      >
        <div
          class="inner"
        >
          <nav>
            <a
              href="#home"
            >
              Home
            </a>
            <a
              href="#work"
            >
              Work
            </a>
            <a
              href="#about"
            >
              About
            </a>
            <a
              href="#contact"
            >
              Contact
            </a>
          </nav>
          <div
            class="socials"
          >
            <svg
              fill="none"
              height="24"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M24 12c0 6.627-5.373 12-12 12S0 18.627 0 12 5.373 0 12 0s12 5.373 12 12z"
                fill="#7289DA"
              />
              <path
                d="M16.135 6.6h-8.27c-.697 0-1.265.552-1.265 1.236v8.112c0 .684.568 1.236 1.265 1.236h6.999l-.327-1.11.79.714.746.672L17.4 18.6V7.836c0-.684-.568-1.236-1.265-1.236zm-2.382 7.836s-.222-.258-.408-.486c.809-.222 1.117-.714 1.117-.714a3.57 3.57 0 0 1-.71.354 4.445 4.445 0 0 1-2.493.252 5.297 5.297 0 0 1-.907-.258 3.669 3.669 0 0 1-.45-.204c-.019-.012-.037-.018-.056-.03a.085.085 0 0 1-.024-.018c-.111-.06-.173-.102-.173-.102s.296.48 1.08.708c-.185.228-.414.498-.414.498-1.364-.042-1.882-.912-1.882-.912 0-1.932.889-3.498.889-3.498.888-.648 1.734-.63 1.734-.63l.062.072c-1.111.312-1.623.786-1.623.786s.135-.072.364-.174c.66-.282 1.184-.36 1.4-.378.038-.006.068-.012.105-.012a5.134 5.134 0 0 1 3.104.564s-.487-.45-1.536-.762l.086-.096s.846-.018 1.735.63c0 0 .888 1.566.888 3.498 0 0-.524.87-1.888.912zm-2.87-2.802c-.352 0-.63.3-.63.666 0 .366.284.666.63.666.352 0 .63-.3.63-.666a.643.643 0 0 0-.63-.666zm2.253 0c-.352 0-.63.3-.63.666 0 .366.284.666.63.666.351 0 .63-.3.63-.666 0-.366-.279-.666-.63-.666z"
                fill="#fff"
              />
            </svg>
            <svg
              fill="none"
              height="24"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M0 12C0 5.373 5.373 0 12 0s12 5.373 12 12-5.373 12-12 12S0 18.627 0 12z"
                fill="#0077B5"
              />
              <path
                clip-rule="evenodd"
                d="M8.66 7.411c0 .785-.591 1.413-1.54 1.413h-.017C6.19 8.824 5.6 8.196 5.6 7.41 5.6 6.611 6.208 6 7.14 6c.93 0 1.503.61 1.52 1.411zM8.48 9.94v8.17H5.761V9.94h2.72zm9.808 8.17v-4.684c0-2.51-1.342-3.678-3.131-3.678-1.444 0-2.09.793-2.451 1.35V9.939h-2.72c.035.767 0 8.171 0 8.171h2.72v-4.563c0-.245.017-.488.089-.663.197-.488.644-.993 1.395-.993.985 0 1.378.75 1.378 1.847v4.372h2.72z"
                fill="#fff"
                fill-rule="evenodd"
              />
            </svg>
            <svg
              fill="none"
              height="24"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M12 0C5.373 0 0 5.508 0 12.304c0 5.436 3.438 10.048 8.206 11.675.6.114.82-.267.82-.592 0-.293-.01-1.263-.016-2.29-3.338.743-4.043-1.452-4.043-1.452-.546-1.423-1.332-1.8-1.332-1.8-1.089-.764.082-.749.082-.749 1.205.087 1.84 1.268 1.84 1.268 1.07 1.881 2.807 1.337 3.491 1.023.108-.795.42-1.338.762-1.645-2.665-.311-5.467-1.366-5.467-6.08 0-1.344.469-2.442 1.236-3.303-.124-.31-.535-1.562.117-3.256 0 0 1.007-.331 3.3 1.26.958-.272 1.984-.409 3.004-.413 1.02.004 2.047.141 3.006.414 2.29-1.592 3.297-1.261 3.297-1.261.653 1.694.242 2.946.118 3.256.77.861 1.235 1.959 1.235 3.302 0 4.726-2.808 5.766-5.48 6.071.43.382.814 1.13.814 2.279 0 1.646-.014 2.97-.014 3.376 0 .327.216.711.825.59C20.566 22.35 24 17.738 24 12.304 24 5.508 18.627 0 12 0z"
                fill="#161514"
              />
              <path
                d="M4.58 17.581c-.026.06-.12.079-.205.037-.087-.04-.135-.122-.107-.183.026-.062.12-.08.206-.038.087.04.137.123.107.184zM5.066 18.131c-.057.054-.17.03-.245-.056-.078-.085-.093-.198-.035-.253.059-.054.167-.029.246.056.078.086.093.199.034.253zM5.538 18.833c-.074.052-.194.003-.268-.105-.073-.108-.073-.238.002-.29.074-.052.192-.006.268.102.073.11.073.24-.002.293zM6.185 19.511c-.066.074-.206.054-.308-.047-.105-.098-.134-.237-.068-.31.066-.075.207-.054.31.046.104.098.136.238.066.311zM7.077 19.904c-.029.096-.163.14-.299.099-.135-.042-.224-.154-.197-.25.029-.096.164-.141.3-.098.136.041.225.152.196.25zM8.058 19.978c.003.1-.112.183-.254.185-.144.003-.26-.078-.261-.177 0-.101.112-.184.256-.186.142-.003.259.078.259.178zM8.97 19.82c.017.098-.082.198-.223.225-.14.026-.268-.035-.286-.132-.017-.1.084-.2.223-.227.141-.025.268.034.286.134z"
                fill="#161514"
              />
            </svg>
          </div>
        </div>
      </div>
    </header>
  </div>
</body>
`;

exports[`matches its snapshot 1`] = `
<body>
  <div>
    <header
      class="mobileHeader"
    >
      <div
        class="logo"
      >
        d.io
      </div>
      <button
        class="menu"
        data-active="false"
        data-testid="hbtn"
        type="button"
      >
        <svg
          fill="none"
          height="24"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <circle
            cx="12"
            cy="12"
            fill="#116ACC"
            r="12"
          />
          <path
            clip-rule="evenodd"
            d="M6 9.333V8h12v1.333H6zm0 3.334h12v-1.334H6v1.334zM8 16h10v-1.333H8V16z"
            fill="#fff"
            fill-rule="evenodd"
          />
        </svg>
      </button>
      <div
        class="mobileMenu"
        data-active="false"
        data-testid="mobileMenu"
      >
        <div
          class="inner"
        >
          <nav>
            <a
              href="#home"
            >
              Home
            </a>
            <a
              href="#work"
            >
              Work
            </a>
            <a
              href="#about"
            >
              About
            </a>
            <a
              href="#contact"
            >
              Contact
            </a>
          </nav>
          <div
            class="socials"
          >
            <svg
              fill="none"
              height="24"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M24 12c0 6.627-5.373 12-12 12S0 18.627 0 12 5.373 0 12 0s12 5.373 12 12z"
                fill="#7289DA"
              />
              <path
                d="M16.135 6.6h-8.27c-.697 0-1.265.552-1.265 1.236v8.112c0 .684.568 1.236 1.265 1.236h6.999l-.327-1.11.79.714.746.672L17.4 18.6V7.836c0-.684-.568-1.236-1.265-1.236zm-2.382 7.836s-.222-.258-.408-.486c.809-.222 1.117-.714 1.117-.714a3.57 3.57 0 0 1-.71.354 4.445 4.445 0 0 1-2.493.252 5.297 5.297 0 0 1-.907-.258 3.669 3.669 0 0 1-.45-.204c-.019-.012-.037-.018-.056-.03a.085.085 0 0 1-.024-.018c-.111-.06-.173-.102-.173-.102s.296.48 1.08.708c-.185.228-.414.498-.414.498-1.364-.042-1.882-.912-1.882-.912 0-1.932.889-3.498.889-3.498.888-.648 1.734-.63 1.734-.63l.062.072c-1.111.312-1.623.786-1.623.786s.135-.072.364-.174c.66-.282 1.184-.36 1.4-.378.038-.006.068-.012.105-.012a5.134 5.134 0 0 1 3.104.564s-.487-.45-1.536-.762l.086-.096s.846-.018 1.735.63c0 0 .888 1.566.888 3.498 0 0-.524.87-1.888.912zm-2.87-2.802c-.352 0-.63.3-.63.666 0 .366.284.666.63.666.352 0 .63-.3.63-.666a.643.643 0 0 0-.63-.666zm2.253 0c-.352 0-.63.3-.63.666 0 .366.284.666.63.666.351 0 .63-.3.63-.666 0-.366-.279-.666-.63-.666z"
                fill="#fff"
              />
            </svg>
            <svg
              fill="none"
              height="24"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M0 12C0 5.373 5.373 0 12 0s12 5.373 12 12-5.373 12-12 12S0 18.627 0 12z"
                fill="#0077B5"
              />
              <path
                clip-rule="evenodd"
                d="M8.66 7.411c0 .785-.591 1.413-1.54 1.413h-.017C6.19 8.824 5.6 8.196 5.6 7.41 5.6 6.611 6.208 6 7.14 6c.93 0 1.503.61 1.52 1.411zM8.48 9.94v8.17H5.761V9.94h2.72zm9.808 8.17v-4.684c0-2.51-1.342-3.678-3.131-3.678-1.444 0-2.09.793-2.451 1.35V9.939h-2.72c.035.767 0 8.171 0 8.171h2.72v-4.563c0-.245.017-.488.089-.663.197-.488.644-.993 1.395-.993.985 0 1.378.75 1.378 1.847v4.372h2.72z"
                fill="#fff"
                fill-rule="evenodd"
              />
            </svg>
            <svg
              fill="none"
              height="24"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M12 0C5.373 0 0 5.508 0 12.304c0 5.436 3.438 10.048 8.206 11.675.6.114.82-.267.82-.592 0-.293-.01-1.263-.016-2.29-3.338.743-4.043-1.452-4.043-1.452-.546-1.423-1.332-1.8-1.332-1.8-1.089-.764.082-.749.082-.749 1.205.087 1.84 1.268 1.84 1.268 1.07 1.881 2.807 1.337 3.491 1.023.108-.795.42-1.338.762-1.645-2.665-.311-5.467-1.366-5.467-6.08 0-1.344.469-2.442 1.236-3.303-.124-.31-.535-1.562.117-3.256 0 0 1.007-.331 3.3 1.26.958-.272 1.984-.409 3.004-.413 1.02.004 2.047.141 3.006.414 2.29-1.592 3.297-1.261 3.297-1.261.653 1.694.242 2.946.118 3.256.77.861 1.235 1.959 1.235 3.302 0 4.726-2.808 5.766-5.48 6.071.43.382.814 1.13.814 2.279 0 1.646-.014 2.97-.014 3.376 0 .327.216.711.825.59C20.566 22.35 24 17.738 24 12.304 24 5.508 18.627 0 12 0z"
                fill="#161514"
              />
              <path
                d="M4.58 17.581c-.026.06-.12.079-.205.037-.087-.04-.135-.122-.107-.183.026-.062.12-.08.206-.038.087.04.137.123.107.184zM5.066 18.131c-.057.054-.17.03-.245-.056-.078-.085-.093-.198-.035-.253.059-.054.167-.029.246.056.078.086.093.199.034.253zM5.538 18.833c-.074.052-.194.003-.268-.105-.073-.108-.073-.238.002-.29.074-.052.192-.006.268.102.073.11.073.24-.002.293zM6.185 19.511c-.066.074-.206.054-.308-.047-.105-.098-.134-.237-.068-.31.066-.075.207-.054.31.046.104.098.136.238.066.311zM7.077 19.904c-.029.096-.163.14-.299.099-.135-.042-.224-.154-.197-.25.029-.096.164-.141.3-.098.136.041.225.152.196.25zM8.058 19.978c.003.1-.112.183-.254.185-.144.003-.26-.078-.261-.177 0-.101.112-.184.256-.186.142-.003.259.078.259.178zM8.97 19.82c.017.098-.082.198-.223.225-.14.026-.268-.035-.286-.132-.017-.1.084-.2.223-.227.141-.025.268.034.286.134z"
                fill="#161514"
              />
            </svg>
          </div>
        </div>
      </div>
    </header>
  </div>
</body>
`;
